lib/mayaUsd/ufe/UsdHierarchy.cpp

```diff
@@ -244,6 +245,23 @@ Ufe::SceneItem::Ptr UsdHierarchy::defaultParent() const
     return createItem(proxyShapePath);
 }
 
+#if UFE_PREVIEW_VERSION_NUM >= 2026
+Ufe::UndoableCommand::Ptr UsdHierarchy::reorderCmd(const Ufe::SceneItemList& orderedList) const
+{
+    std::vector<TfToken> orderedTokens;
+
+    for (const auto& item : orderedList) {
+        orderedTokens.emplace_back(downcast(item)->prim().GetPath().GetNameToken());
+    }
+
+    // TODO HS Oct 23, 2020, grab any child and pass it to UsdUndoReorderCommand
```

Comment:
@ppt-adsk 
But `downcast(sceneItem()) ` doesn't give me the parent and rather the item that I am moving.

```
e.g
/Xform1/pSphere1
```

What I want is the parent (/Xform1) since that's the one whose spec is going to call the SetNameChildrenOrder. 

see 0977cf12e569a8eec7d8b05982d832f112846022

Reply:
Hmm, I'm confused!  Isn't this a method that is called on the parent, to reorder its children?  I.e. you're calling it on /Xform1's hierarchy interface, to reorder its children, so Hierarchy::sceneItem() will in fact give you the parent.

Reply:
Ahhh.. sorry about the confusion. Now I see it... :) I was calling the the command on a wrong hierarchy interface.  

I have it fixed soon on the data model side and then make this change.

Reply:
@ppt-adsk addressed in 520b36de46f29948c6fc3ad99a67f2080c3f1cda